plugins/module_utils/_hashi_vault_common.py

```diff
@@ -57,6 +61,20 @@ def get_vault_client(
         :type hashi_vault_revoke_on_logout: bool
         '''
 
+        url = kwargs.get('url') or os.environ.get('VAULT_ADDR')
```

Comment:
I think we need to remove the env var retrieval here. This collection specially processes some env vars and `VAULT_ADDR` is one of them. I _think_ this method will happen after env late-binding so hopefully we can just remove the `or` condition here.

See also:
- https://github.com/ansible-collections/community.hashi_vault/blob/main/plugins/module_utils/_hashi_vault_common.py#L203
- https://github.com/ansible-collections/community.hashi_vault/blob/main/plugins/module_utils/_connection_options.py#L68